File: Android/app/src/main/kotlin/com/zhechengqi/tollcat/developer/DashboardLabSection.kt
package com.zhechengqi.tollcat.developer

import com.zhechengqi.tollcat.R
import com.zhechengqi.tollcat.dashboard.DashboardModules

/** 实验室目录的分组。跟编辑面同一份模块清单，不另开名单。 */
enum class DashboardLabSection {
    Hero,
    Attention,
    Optional,
    ;

    val titleRes: Int
        get() = when (this) {
            Hero -> R.string.dev_lab_section_hero
            Attention -> R.string.dashboard_attention
            Optional -> R.string.dev_lab_section_optional
        }

    val ids: List<String>
        get() = DashboardLabModules.defaultOrder.filter { section(it) == this }

    companion object {
        fun section(id: String): DashboardLabSection = when {
            id == DashboardLabModules.MONTH_TO_DATE || id in DashboardModules.fixedSlot -> Hero
            id in DashboardModules.attention -> Attention
            else -> Optional
        }
    }
}

File: Packages/MeterKit/Sources/MeterFeatures/Developer/DashboardLabSection.swift
#if DEBUG
import Foundation
import MeterModules

/// 实验室目录的分组。跟编辑面同一份模块清单，不另开名单。
enum DashboardLabSection: String, CaseIterable, Identifiable {
    case hero
    case attention
    case optional

    var id: String { rawValue }

    var title: LocalizedStringResource {
        switch self {
        case .hero: L("英雄区")
        case .attention: L("需要注意")
        case .optional: L("可选项")
        }
    }

    var ids: [DashboardModuleID] {
        DashboardModuleID.defaultOrder.filter { Self.section(for: $0) == self }
    }

    static func section(for id: DashboardModuleID) -> DashboardLabSection {
        if id.isPinned || id.isFixedSlot { return .hero }
        if id.isAttention { return .attention }
        return .optional
    }
}
#endif
